### PR TITLE
Add dropout

### PR DESCRIPTION
**Pull request by collin-k**
**Description**
This PR introduces dropout to our three models. This feature is a crucial part of our project's roadmap as it addresses issues with overfitting.

- Added dropout parameter to model config
- Created new fcn module to implement dropout
- Added dropout argument to "Unet," "DeepLabV3Plus," and "FCN" models
- Added if-statement in training phase to check if model(x) returns a tuple. This addresses the issue that occurs when using the "Unet" and "DeepLabV3Plus" models as model(x) returns a tuple when dropout is implemented.

### CODE REVIEW

**Review by annawoodard — APPROVED** [review]
Next time please omit unrelated changes, e.g. LR change, tile size change, pre commit change. Otherwise, great, thank you!
